### PR TITLE
[munin-httpd] add host option

### LINKED ISSUE #1089: munin-httpd bind to 127.0.0.1
**Is your feature request related to a problem? Please describe.**
 munin 2.999.8 with lighttpd as proxy, but I could not bind munin-httpd to 127.0.0.1 
**Describe the solution you'd like**
setup munin-httpd bind to 127.0.0.1  currently it is on :::4948

### PR DESCRIPTION
**Pull request by cgzones**
closes #1089

### CODE REVIEW

**Comment by cgzones** [comment]
updated with `--limit` option

File: script/munin-httpd [hunk 1]
```diff
@@ -35,21 +35,21 @@ package main;    ## no critic qw(Modules::ProhibitMultiplePackages)
...
-    "port=i"  => \$port,
+    "listen=s"  => \$listen,
     "version" => \&print_version_and_exit,
     "help"    => \&print_usage_and_exit,
   );
 
-# un-taint port number
-die "ERROR: Invalid port '$port' supplied.\n" unless $port =~ /^(\d+)$/x;
-$port = $1;
+die "ERROR: Invalid listen argument supplied: '$listen'\n" unless $listen =~ /^(\S+)?:(\d+)?$/x;
```
Comment (sumpfralle) [inline script/munin-httpd]: Does this work for IPv6? :)

File: script/munin-httpd [hunk 2]
```diff
@@ -35,21 +35,21 @@ package main;    ## no critic qw(Modules::ProhibitMultiplePackages)
...
+my $host = defined $1 ? $1 : "*";
+my $port = defined $2 ? $2 : 4948;
```
Comment (sumpfralle) [inline script/munin-httpd]: Here you are using an integer (the default) or a string (parsed from the argument). Is both acceptable for `Http->new`?

File: script/munin-httpd [hunk 3]
```diff
@@ -35,21 +35,21 @@ package main;    ## no critic qw(Modules::ProhibitMultiplePackages)
...
+    "listen=s"  => \$listen,
     "version" => \&print_version_and_exit,
     "help"    => \&print_usage_and_exit,
   );
 
-# un-taint port number
-die "ERROR: Invalid port '$port' supplied.\n" unless $port =~ /^(\d+)$/x;
-$port = $1;
+die "ERROR: Invalid listen argument supplied: '$listen'\n" unless $listen =~ /^(\S+)?:(\d+)?$/x;
+my $host = defined $1 ? $1 : "*";
```
Comment (sumpfralle) [inline script/munin-httpd]: I would tend to default to localhost here.
(or maybe the empty string, if `Http->new` supports that)

File: script/munin-httpd (hunk 1 above)
Comment (cgzones) [reply script/munin-httpd]: now :-)

File: script/munin-httpd (hunk 2 above)
Comment (cgzones) [reply script/munin-httpd]: yes

File: script/munin-httpd (hunk 3 above)
Comment (cgzones) [reply script/munin-httpd]: "*" was the previous default

File: script/munin-httpd [hunk 4]
```diff
@@ -35,21 +35,21 @@ package main;    ## no critic qw(Modules::ProhibitMultiplePackages)
...
-    "port=i"  => \$port,
+    "listen=s"  => \$listen,
     "version" => \&print_version_and_exit,
     "help"    => \&print_usage_and_exit,
   );
 
-# un-taint port number
-die "ERROR: Invalid port '$port' supplied.\n" unless $port =~ /^(\d+)$/x;
-$port = $1;
+die "ERROR: Invalid listen argument supplied: '$listen'\n" unless $listen =~ /^(\S+|\[\S+\])?:(\d+)?$/x;
```
Comment (steveschnepp) [inline script/munin-httpd]: Waaaa... My head hurts reading regex 💥

File: script/munin-httpd [hunk 5]
```diff
@@ -93,9 +93,9 @@ It supports three click zooming.
 
 =over 5
 
-=item B<< --port <portnumber> >>
+=item B<< --listen <hostname:port> >>
 
-Sets the port number to run on. [4948]
+Sets the hostname and port to listen on. [*:4948]
```
Comment (steveschnepp) [inline script/munin-httpd]: Perfect. It would be nice to have the same level of support as the Apache `Listen`
https://httpd.apache.org/docs/2.4/bind.html

**Pull request merged**